src/pages/index.js

```diff
@@ -5,52 +5,348 @@ import PopupWithForm from "../components/PopupWithForm.js";
+      })
+      .catch((err) => console.log(err));
+  } else {
+    api
+      .addLike(cardData._id)
+      .then((updatedCard) => {
+        cardData.isLiked = true; // Update local like status
+        updateCardLikes(cardData._id, updatedCard.likes); // Update UI with new like count
+      })
+      .catch((err) => console.log(err));
+  }
+}
+
+// Function to update card likes in the UI
+// function updateCardLikes(cardId, likes) {
+//   const cardElement = document.querySelector(`.card[data-id="${cardId}"]`); // Make sure card elements have data-id attribute
+//   const likeCountElement = cardElement.querySelector(".card__like-count");
+//   likeCountElement.textContent = likes.length;
+
+//   const likeButton = cardElement.querySelector(".card__button-like");
+//   if (likes.some((user) => user._id === currentUser._id)) {
+//     likeButton.classList.add("card__button-like_active");
+//   } else {
+//     likeButton.classList.remove("card__button-like_active");
+//   }
+// }
+
+// // Attach event listeners to like buttons
+// document.addEventListener("click", (event) => {
+//   if (event.target.classList.contains("card__button-like")) {
+//     const cardElement = event.target.closest(".card");
+//     const cardId = cardElement.dataset.id;
+//     const cardData = findCardById(cardId);
+//     console.log("Like button clicked for card ID:", cardId); // Debugging log
+//     console.log("Card data for like:", cardData);
+//     // const cardData = findCardById(cardId); // Implement this function to find the card object
+//     handleLikeClick(cardData);
+//   }
+// });
+
+// ----------------------------
+
+const profileSubmitButton = document.querySelector("#profile-submit-button");
+
+function handleProfileFormSubmit(event) {
+  // event.preventDefault();
+  profileSubmitButton.textContent = "Saving...";
+
+  const formData = new FormData(profileForm);
+
+  const formValues = {
+    name: formData.get("title"),
+    about: formData.get("description"),
+  };
+
+  api
+    .setUserInfo(formValues)
+    .then((data) => {
+      userInfo.setUserInfo({ name: data.name, job: data.about });
+      editProfilePopup.close();
+    })
+    .catch((error) => console.error("Error updating profile:", error))
+    .finally(() => {
+      profileSubmitButton.textContent = "Save";
+    });
+}
+
+profileForm.addEventListener("submit", handleProfileFormSubmit);
 
-const handleAddCardFormSubmit = (data) => {
-  renderCard({ name: data.title, link: data.url });
-  newCardForm.reset();
-  newCardPopup.close();
-  newCardFormValidator.toggleButtonState();
-};
+// ----------------------------
+const addCardSubmitButton = document.querySelector("#add-card-submit-button");
+
+function handleAddCardFormSubmit(event) {
+  // event.preventDefault();
+  addCardSubmitButton.textContent = "Saving...";
+
+  const addCardForm = document.querySelector("#add-card-form");
+  const title = addCardForm.elements.title.value;
+  const url = addCardForm.elements.url.value;
```

Comment:
Only method `_getInputValues` collects the whole data of the inputs **inside** `PopupWithForm` and passes it to the call of the `submit handler`. So you need to use the collected data rather than the inputs.

Reply:
i don't understand. can you please elaborate?

Reply:
> i don't understand. can you please elaborate?

You have method `_getInputValues` inside `PopupWithForm` which collects the whole data from the inputs. You get this data in the call


```js
function handleAddCardFormSubmit(data) {


```